test/framework/integration_framework/integration_test_framework.hpp

```diff
@@ -52,40 +50,131 @@ namespace integration_framework {
     using BlockType = std::shared_ptr<shared_model::interface::Block>;
 
    public:
-    IntegrationTestFramework(size_t maximum_block_size = 10)
-        : maximum_block_size_(maximum_block_size) {}
+    /**
+     * Construct test framework instance
+     * @param maximum_proposal_size - (default = 10) Maximum amount of
+     * transactions per proposal
+     * @param destructor_lambda - (default nullptr) Pointer to function which
+     * receives pointer to constructed instance of Integration Test Framework.
+     * If specified, then will be called instead of default destructor's code
+     */
+    explicit IntegrationTestFramework(
+        size_t maximum_proposal_size = 10,
+        void (*destructor_lambda)(IntegrationTestFramework *) = nullptr)
+        : maximum_proposal_size_(maximum_proposal_size),
+          destructor_lambda_(destructor_lambda) {}
+
+    /**
+     * Initialize Iroha instance with default genesis block and provided signing
```

Comment:
Default block generation will be moved to a separate function, but two versions of setInitialState are about to exist (because defaultBlock also uses keypair as an argument).